src/main/java/teammates/ui/webapi/GetFeedbackSessionsAction.java

```diff
@@ -109,7 +143,14 @@ public JsonResult execute() {
 
         FeedbackSessionsData responseData = new FeedbackSessionsData(feedbackSessionAttributes);
         if (entityType.equals(Const.EntityType.STUDENT)) {
-            responseData.getFeedbackSessions().forEach(FeedbackSessionData::hideInformationForStudent);
+            String emailAddress = students.isEmpty() ? null : students.get(0).getEmail();
+            responseData.getFeedbackSessions().forEach(session -> {
+                session.hideInformationForStudent();
+                if (emailAddress == null) {
+                    return;
+                }
+                session.filterDeadlinesForStudent(emailAddress);
```

Comment:
A little redundant because this cannot happen. Logically, this would not make sense too since the entityType is STUDENT and we gotta have some student

Reply:
I see. I was not sure if it was completely guaranteed. I was concerned that `logic.getXyzsForGoogleId` could possibly return an empty list.

Reply:
Yup. Though your logic before kind of prevents it already.

Reply:
Actually looking at it again now, it must not be empty by this point because if it were, the ENFEs would have been thrown, and it would not reach that point in the first place.

Reply:
Small note here @jayasting98, try to avoid throwing student not found exception unless necessary. The old contract of this API is to return empty list if no students found. There's a good chance things break if we change this to throw exception instead.

Reply:
Yes. Then again, there's no telling if the logic above would be changed someday. But for now, if the GoogleID doesn't get you a user, it is totally ok to just throw. Maybe making an assert would be more appropriate than returning it silently?

Reply:
A side note here, it doesn't really make sense for there to be a possible path for students where the deadlines are not filtered out, unless missing student entity somehow means that the user has higher level of permissions, which doesn't seem right.

Reply:
Yes... the odd feeling I got from this early return is perhaps from this. We should just assert that the user exist and leave it to the future dev to resolve it should the absence of user make sense at this point of logic

Reply:
Ok I will remove the ENFEs. Anyway, looking at the access control, I think because the user must be logged in, then the user must be guaranteed to exist. Hopefully, this is correct.

Reply:
Adding assert here will probably be harmless as the current logic stands🤔

Reply:
"I think because the user must be logged in, then the user must be guaranteed to exist". Distracted so can't verify this at the moment, but in general this is not true. Account entity exists does not imply student/instructor entity exists.

Reply:
>Account entity exists does not imply student/instructor entity exists

I think you're right, it does not, but in the specific access control, it also checks if `isStudent` is true for `EntityType.STUDENT`, same for instructors. `isStudent` is only marked true if there are actually students with that Google ID in the database, so it should be all good, right?

Reply:
>Adding assert here will probably be harmless as the current logic stands

Sure, I can add it too.

Reply:
I haven't looked into it, but trivially if user is admin the specific access control method returns early.

Reply:
Yes, but currently, there are no requests for this endpoint from the admin pages if I'm not wrong.

Reply:
Let's not waste time on this, it's not very relevant to PR at hand at the moment. Just need to be aware that we cannot assume anything about the existence of Student/Instructor entity from the user info alone.

Reply:
Sorry, just scared of bugs, but I will leave it be.